spark/v4.0/spark/src/main/java/org/apache/iceberg/spark/SparkWriteConf.java

```diff
@@ -461,11 +461,22 @@ public boolean caseSensitive() {
         .parse();
   }
 
+  private String getWapBranch() {
+    return confParser.stringConf().sessionConf(SparkSQLProperties.WAP_BRANCH).parseOptional();
+  }
+
+  public boolean isWapBranch(String branchName) {
```

Comment:
I'm not sure why we need this method? Aren't we just always returning true of the wapBranch is not null and matches the branch being written too?